scripts/globals/effects/costume.lua

```diff
@@ -4,25 +4,27 @@
 --
 -----------------------------------
 
------------------------------------
--- onEffectGain Action
------------------------------------
-
 function onEffectGain(target,effect)
-    target:costume(effect:getPower());
-end;
-
------------------------------------
--- onEffectTick Action
------------------------------------
+    target:costume(effect:getPower())
+    if effect:getPower() == 1999 then
+        if effect:getTimeRemaining() / 1000 >= 7170 then
+            target:speed(target:speed()+5)
+            effect:setPower(19991)
+        else
+            target:speed(target:speed()+10)
+            effect:setPower(19992)
+        end
+    end
+end
 
 function onEffectTick(target,effect)
-end;
-
------------------------------------
--- onEffectLose Action
------------------------------------
+end
 
 function onEffectLose(target,effect)
-    target:costume(0);
-end;
\ No newline at end of file
+    target:costume(0)
+    if effect:getPower() == 19991 then
+        target:speed(target:speed()-5)
```

Comment:
almost anyplace you think youw nat to use `speed()` you will wanna use modifier for move instead, so we can addMod/delMod on effect gain/loss. You can't trust that the speed hasn't been changed again between adding and losing the effect, could gum it up.